wgpu-hal/src/dx12/mod.rs

```diff
@@ -546,6 +546,12 @@ pub struct Surface {
 unsafe impl Send for Surface {}
 unsafe impl Sync for Surface {}
 
+impl Surface {
+    pub fn swap_chain(&self) -> &RwLock<Option<SwapChain>> {
+        &self.swap_chain
+    }
```

Comment:
Instead of needing to expose SwapChain and bounce through it to get the swapchain, you should be able to return `Option<dxgi::IDXGISwapChain3>` by doing something like:

```suggestion
    pub fn swap_chain(&self) -> Option<Dxgi::IDXGISwapChain3> {
        Some(self.swap_chain.lock().as_ref()?.raw.clone())
    }
```

Reply:
Totally fine with that!